pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -352,58 +352,90 @@ func getServerByName(client *gophercloud.ServiceClient, name string) (*servers.S
 	return &serverList[0], nil
 }
 
-func findAddrs(netblob interface{}) []string {
-	// Run-time types for the win :(
-	ret := []string{}
-	list, ok := netblob.([]interface{})
-	if !ok {
-		return ret
-	}
-	for _, item := range list {
-		props, ok := item.(map[string]interface{})
-		if !ok {
-			continue
-		}
-		tmp, ok := props["addr"]
+func getAddressesByName(client *gophercloud.ServiceClient, name string) ([]api.NodeAddress, error) {
+	srv, err := getServerByName(client, name)
+	if err != nil {
+		return nil, err
+	}
+
+	addrs := []api.NodeAddress{}
+
+	for network, netblob := range srv.Addresses {
+		var addressType api.NodeAddressType
```

Comment:
Move this declaration down into the `for _, item := range list { ... }` block below where it is used - just so we can be sure we aren't mistakenly carrying over `addressType` from some earlier iteration of the below loop.